uwrt_utils/include/uwrt_utils/uwrt_can.h

```diff
@@ -0,0 +1,102 @@
+
+#include <linux/can.h>
+#include <linux/can/raw.h>
+#include <net/if.h>
+#include <chrono>
+#include <map>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <vector>
+
+namespace uwrt_can {
+
+class UWRTCANWrapper {
+ public:
+  explicit UWRTCANWrapper() = default;
+  explicit UWRTCANWrapper(std::string name, std::string interface_name, int thread_sleep_millis = 1)
+      : name_(std::move(name)),
+        interface_name_(std::move(interface_name)),
+        thread_sleep_millis_(std::chrono::milliseconds(thread_sleep_millis)){};
+  // TODO (wraftus) make other constructor for failover CAN
+
+  explicit UWRTCANWrapper(const UWRTCANWrapper& to_copy) = delete;
+  explicit UWRTCANWrapper(UWRTCANWrapper&& to_move) noexcept
+      : name_(std::move(to_move.name_)),
+        interface_name_(std::move(to_move.interface_name_)),
+        thread_sleep_millis_(to_move.thread_sleep_millis_) {
+    to_move.read_thread_running_ = false;
+    if (to_move.read_thread_.joinable()) {
+      to_move.read_thread_.join();
+    }
+  }
+
+  UWRTCANWrapper& operator=(const UWRTCANWrapper& to_copy) = delete;
+  UWRTCANWrapper& operator=(UWRTCANWrapper&& to_move) noexcept {
+    name_ = std::move(to_move.name_);
+    interface_name_ = (std::move(to_move.interface_name_));
+    thread_sleep_millis_ = to_move.thread_sleep_millis_;
+    initialized_ = false;
+
+    to_move.read_thread_running_ = false;
+    if (to_move.read_thread_.joinable()) {
+      to_move.read_thread_.join();
+    }
+    read_thread_running_ = false;
+    if (read_thread_.joinable()) {
+      read_thread_.join();
+    }
+    return *this;
+  }
+
+  ~UWRTCANWrapper() {
+    read_thread_running_ = false;
+    if (read_thread_.joinable()) {
+      read_thread_.join();
+    }
+  }
+
+  enum class UWRTCANStatus {
```

Comment:
see comment above. i think its better to use exceptions instead of return codes.